### ISSUE TITLE
about moving average

### ISSUE THREAD

**Issue by hityzy** [duxingren14/DualGAN#3]
I see that you add tf.contrib.layers.batch_norm in the networks, but I can't fine where do you update the moving average of bn

**Comment by duxingren14** [comment]
The self.d_optim and, self.g_optim will do it automatically. Please look at
how d_vars and g_vars are defined.
Thanks.

Zili

**Comment by hityzy** [comment]
Thanks for your replay! Maybe I have made a mistake, however it is said in "https://www.tensorflow.org/api_docs/python/tf/contrib/layers/batch_norm" that we should add update ops before optimizer like this:
 update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
  with tf.control_dependencies(update_ops):
    train_op = optimizer.minimize(loss)
 
Maybe the vars for derivation may not contain moving average, because moving average is not updated by SGD.

**Comment by duxingren14** [comment]
I am sure moving average is a member of tf.trainable_variables.

What I understand about the document is, the SGD may update the moving
average. However, that can have a speed penalty, especially in distributed
settings. Maybe the UPDATE_OPS have been optimised for speed.
